gcp/public-agent.tf

```diff
@@ -0,0 +1,191 @@
+
+  instances = ["${google_compute_instance.public-agent.*.self_link}"]
+
+  health_checks = [
+    "${google_compute_http_health_check.public-agent-adminrouter-healthcheck.name}"
+  ]
+}
+
+# Used for the internal load balancer. The external load balancer only supports google_compute_http_health_check resource.
+resource "google_compute_health_check" "public-agent-healthcheck" {
+  name               = "${data.template_file.cluster-name.rendered}-mesos-public-agent-healthcheck"
+  check_interval_sec = 30
+  timeout_sec        = 5
+  healthy_threshold = 2
+  unhealthy_threshold = 2
+
+  http_health_check {
+    port = "5050"
+  }
+}
+
+# Used for the external load balancer. The external load balancer only supports google_compute_http_health_check resource.
+resource "google_compute_http_health_check" "public-agent-adminrouter-healthcheck" {
+  name                = "${data.template_file.cluster-name.rendered}-external-mesos-http-public-agent-healthcheck"
+  check_interval_sec  = 30
+  timeout_sec         = 5
+  healthy_threshold   = 2
+  unhealthy_threshold = 2
+  port                = "80"
+
+  # depends_on = ["google_compute_target_pool.public-agent-pool"]
+}
+
+resource "google_compute_instance_group" "public-agent" {
+  name        = "${data.template_file.cluster-name.rendered}-public-agent-cluster"
+  description = "DC/OS Public Agent Instance Group"
+
+  instances = [
+    "${google_compute_instance.public-agent.*.self_link}"
+  ]
+
+  named_port {
+    name = "http"
+    port = "80"
+  }
+
+  named_port {
+    name = "https"
+    port = "443"
+  }
+
+  named_port {
+    name = "mesos-public-agent"
+    port = "5051"
+  }
+
+  named_port {
+    name = "ssh"
+    port = "22"
+  }
+
+  zone = "${data.google_compute_zones.available.names[0]}"
+}
+
+# deploy image
+resource "google_compute_instance" "public-agent" {
+   name         = "${data.template_file.cluster-name.rendered}-public-agent-${count.index + 1}"
+   machine_type = "${var.gcp_public_agent_instance_type}"
+   zone         = "${data.google_compute_zones.available.names[0]}"
+   count        = "${var.num_of_public_agents}"
+ 
+  labels {
+   owner = "${coalesce(var.owner, data.external.whoami.result["owner"])}"
+   expiration = "${var.expiration}"
+   name = "${data.template_file.cluster-name.rendered}-public-agent-${count.index + 1}"
+   cluster = "${data.template_file.cluster-name.rendered}"
+  }
+
+  boot_disk {
+    initialize_params {
+      image = "${module.dcos-tested-gcp-oses.gcp_image_family}/${module.dcos-tested-gcp-oses.gcp_image_name}"
+      size  = "${var.instance_disk_size}"
+    }
+  }
+
+  connection {
+    # host = "${element(google_compute_instance.public-agent.*.network_interface.0.access_config.0.assigned_nat_ip, count.index)}"
+    user = "${coalesce(var.gce_ssh_user, module.dcos-tested-gcp-oses.user)}"
+  }
+
+  network_interface {
+    subnetwork = "${google_compute_subnetwork.public.name}"
+    access_config {
+        // IP
```

Comment:
remove the comment or add meaningful message

Reply:
done! tyvm